apps/backend/src/factory/xlsx/callFaps.ts

```diff
@@ -0,0 +1,174 @@
+import { stripHtml } from 'string-strip-html';
+import { container } from 'tsyringe';
+
+import baseContext from '../../buildContext';
+import { Tokens } from '../../config/Tokens';
+import { FapDataSource } from '../../datasources/FapDataSource';
+import { ProposalEndStatus } from '../../models/Proposal';
+import { UserWithRole } from '../../models/User';
+import { RowObj, collectFapXLSXRowData } from './fap';
+
+const fapDataSource: FapDataSource = container.resolve(Tokens.FapDataSource);
+
+const ProposalEndStatusStringValue = {
+  [ProposalEndStatus.UNSET]: 'Unset',
+  [ProposalEndStatus.ACCEPTED]: 'Accepted',
+  [ProposalEndStatus.RESERVED]: 'Reserved',
+  [ProposalEndStatus.REJECTED]: 'Rejected',
+};
+
+export type CallRowObj = RowObj & {
+  fapMeetingDecision?: string | null;
+  fapMeetingExComment?: string | null;
+  fapMeetingInComment?: string | null;
+};
+
+export const collectCallFapXLSXData = async (
+  callId: number,
+  user: UserWithRole
+) => {
+  const faps = await baseContext.queries.fap.dataSource.getFapsByCallId(callId);
+  const call = await baseContext.queries.call.get(user, callId);
+  const filename = `${call?.shortCode}_FAP_Results.xlsx`;
+
+  const baseData = await Promise.all(
+    faps.map(async (fap) => {
+      return {
+        sheetName: fap.code,
+        rows: await collectFAPRowData(fap.id, callId, user),
```

Comment:
I see that this `collectFAPRowData` function is used before it is defined. Maybe it is good to move it on the top.